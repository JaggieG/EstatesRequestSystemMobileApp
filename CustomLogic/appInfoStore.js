import { createStore} from 'redux'
import { updateAppInfo } from './storage.js'

// API URLs
const api_server_url_testing = "http://jaglocaltesttemp.aiglon.ch:8080"
const api_server_url_prod = "https://webapp-googleclassroomadmin-staging.azurewebsites.net"
const api_server_url = api_server_url_testing
const api_path = "/genericsolutions/estatesrequestsystem/mobile_api/"
const authentication_endpoint = "authMobileDevice"
const retrieval_endpoint = "retrieval"

const defaultState = {
    loginSuccess: false,
    email_address : null,
    display_name : null,
    JWT_Token : null,
    app_language : 'en',
    api_details :  {
          api_server_url : api_server_url,
          api_path : api_path,
          authentication_endpoint : authentication_endpoint,
          retrieval_endpoint : retrieval_endpoint,
     }
  }

function appInfo(state=defaultState, action) {
  switch(action.type) {
    case "LOGIN":
        var new_state = {...state,
                        email_address: action.payload.email_address, 
                        display_name: action.payload.display_name, 
                        JWT_Token: action.payload.JWT_Token, 
                        loginSuccess: true,
        }
        updateAppInfo(JSON.stringify(new_state))
        return new_state;
     case "LOGOUT":
        var new_state = {...state,
                        email_address: null, 
                        display_name: null, 
                        JWT_Token: null, 
                        loginSuccess: false,
          }
          updateAppInfo(JSON.stringify(new_state))
        return new_state ;
      case "STARTUP":
        var new_state = {...state,
                        email_address: action.payload.email_address, 
                        display_name: action.payload.display_name, 
                        JWT_Token: action.payload.JWT_Token, 
                        loginSuccess: true,
          }
          updateAppInfo(JSON.stringify(new_state))
          return new_state;
       case "LANG_UPDATE":
            var new_state = {...state,
                app_language: action.payload.app_language, 
              }
              updateAppInfo(JSON.stringify(new_state))
              return new_state;
     default:
       return state
  }
}

export default createStore(appInfo);