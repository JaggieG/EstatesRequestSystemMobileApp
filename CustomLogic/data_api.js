export const getAllMyRequests = async (appInfo, boolCompleted, callback) => {
    console.log('boolCompleted in data api: ' + boolCompleted)
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getMyRequests'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {
                    boolCompleted : boolCompleted,
                }
            })//
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}

export const getAllMyAssignedRequests = async (appInfo, boolCompleted, callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getAssignedRequests'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {
                    boolCompleted : boolCompleted,
                }
            })//
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}


export const getRequestRecordCounts = async (appInfo, boolCompleted, callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getRequestRecordCounts'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {
                    boolCompleted : boolCompleted,
                }
            })//
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}



export const createANewRequst = async (appInfo, 
                                        building,
                                        building_id,
                                        urgency,
                                        urgency_id, 
                                        description,
                                        title,
                                        callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'createARequest'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)

    
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },

            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {
                    building: building,
                    building_id: building_id,
                    urgency: urgency,
                    urgency_id: urgency_id,
                    description: description,
                    title : title,
                    userDisplayName : appInfo.display_name
                }
            })

        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}

export const getUrgencyList = async (appInfo, callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getUrgency'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)

    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {

                }
            })
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}

export const getItemList = async (appInfo, callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getItems'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {

                }
            })//
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}

export const getSystemRole = async (appInfo, callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getSystemRole'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)
    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                authObject : {
                    email_address: email_address
                },
                payload : {

                }
            })//
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }
}

const getBaseURL = (appInfo)  => {
    return appInfo.api_details.api_server_url + appInfo.api_details.api_path
}

const getEmailAddress = (appInfo) => {
    return appInfo.email_address
}

const getJWTToken = (appInfo) => {
    return appInfo.JWT_Token
}