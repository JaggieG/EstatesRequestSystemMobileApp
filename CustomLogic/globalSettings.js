export const forceDevCreditionals = true
export const forced_dev_emailAdddress = 'jag@aiglon.ch'
export const forced_dev_displayName = 'John Gerhardt'
export const forced_dev_JWTToken = 'FORCED_DEV'


export const getCurrentActiveLanguage = (appInfoStore) => {
    if (appInfoStore == undefined) {
        return 'en'
    } else {
        var appinfo = appInfoStore.getState()
        return appinfo.app_language
    }
    
}


// API URLs
const api_server_url_testing = "http://jaglocaltesttemp.aiglon.ch:8080"
const api_server_url_testing_no_auth = "http://192.168.1.128:8080"
const api_server_url_prod = "https://webapp-googleclassroomadmin-staging.azurewebsites.net"
const api_server_url = api_server_url_testing_no_auth
const api_path = "/genericsolutions/estatesrequestsystem/mobile_api/"
const authentication_endpoint = "authMobileDevice"

export const defaultAppInfo = {
    appInfo : {
      email_address : null,
      display_name : null,
      JWT_Token : null,
      api_details :  {
            api_server_url : api_server_url,
            api_path : api_path,
            authentication_endpoint : authentication_endpoint
       }
     }
    }
