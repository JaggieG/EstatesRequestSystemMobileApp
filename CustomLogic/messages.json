{
    "allMessages" : {
        "my_requests_tab" : {
            "en" : "My Requests",
            "fr" : "Mes Requêtes"
        },
        "make_request_tab" : {
            "en" : "Make a Request",
            "fr" : "Créer une Requête"
        },
        "assigned_requests_tab" : {
            "en" : "Assigned",
            "fr" : "Attribué"
        },
        "connection_tab" : {
            "en" : "Connection",
            "fr" : "Connexion"
        },
        "settings_tab" : {
            "en" : "Settings",
            "fr" : "Réglages"
        },

        

    "assigned_reqeust" : {
        "en" : "Assigned Requests",
        "fr" : "Requêtes attribuées"
    },


        "not_logged_in" : {
            "en" : "You are not currently logged in \r\n\r\n Please click below to connect with your Google Workspace account",
            "fr" : "Vous n'êtes pas connecté, \r\n\r\n Veuillez cliquer le bouton ci-dessous pour se connecter à votre compte Google Workspace"
        },
        "logged_in" : {
            "en" : "You are currently logged in\r\n\r\n Please see details below",
            "fr" : "Vous êtes connecté\r\n Détails ci-dessous"
        },
        "log_in_again" : {
            "en" : "Click the button in order to refresh the connection",
            "fr" : "Cliquer le bouton pour vous connecter à nouveau."
        },
        "choose_language" : {
            "en" : "Choose Language",
            "fr" : "Choisir Langue."
        },


        "authentication_failure" : {
            "en" : "Oops something went wrong with authentication! - Please speak to the IT",
            "fr" : "Une erreur s'est produite, veuillez contacter le service informatique"
        },
        "urlFailure" : {
            "en" : "In order to be able to call the oAuth we need a redirect URL - this is taken when the expo app is opened with the barcode, please scan a barcode",
            "fr" : "Pour pouvoir appeler oAuth, nous avons besoin d'une redirection d'URL - celle-ci est prise lorsque l'application expo est ouverte avec le code-barres, veuillez scanner un code-barres"
        },
        "forced_dev" : {
            "en" : "You have been forced into DEV mode",
            "fr" : "Vous êtes forcé dans le mode développement"
        },
        "authentication_failure_noRecords" : {
            "en" : "No records were returned from the auth server, bizarre! Please speak to IT",
            "fr" : "Aucune entrée n'a été retournée du serveur d'authentification! Veuillez contacter la service informatique"
        },


        "make_request_building" : {
            "en" : "Building / Location",
            "fr" : "Bâtiment / Endroit"
        },
        "make_request_urgency" : {
            "en" : "Urgency",
            "fr" : "Urgence"
        },
        "make_request_description" : {
            "en" : "Description",
            "fr" : "Description"
        },
        "make_request_title" : {
            "en" : "Title",
            "fr" : "Titre"
        },
        "assigned_to" : {
            "en" : "Assigned To",
            "fr" : "Assigner à"
        },

        "desc_en" : {
            "en" : "English Desc",
            "fr" : "Desc Anglais"
        },

        "desc_fr" : {
            "en" : "French Desc",
            "fr" : "Desc Français"
        },

        "submitted_data_time" : {
            "en" : "Submitted @",
            "fr" : "Soumis à"
        },

        "creator_email" : {
            "en" : "Creator Email",
            "fr" : "Créer Email "
        },

        "creator_name" : {
            "en" : "Creator Name",
            "fr" : "Créer Nom"
        },
        
        
        "assigned_to_email" : {
            "en" : "Assigned to Email",
            "fr" : "Attribué à l'email"
        },
        
        "completed" : {
            "en" : "Completed?",
            "fr" : "Fini?"
        },

        "close_request" : {
            "en" : "Close Request",
            "fr" : "Ferme demande"
        },

        "yes" : {
            "en" : "Yes",
            "fr" : "Oui"
        },
        "no" : {
            "en" : "No",
            "fr" : "Non"
        },
    
        "sure_close" : {
            "en" : "Are you sure you want to close this request?",
            "fr" : "Êtes-vous sures de vouloir fermer cette demande?"
        },

        

        

        

        "no_records_found" : {
            "en" : "No Records Found",
            "fr" : "Pas d'informations trouvées"
        },

        


        "please_choose" : {
            "en" : "Tap to choose",
            "fr" : "Veuillez Choisir"
        },
        "submit" : {
            "en" : "Send",
            "fr" : "Envoyer"
        },

        "hide_modal" : {
            "en" : "OK",
            "fr" : "OK"
        },



        "make_request_no_building" : {
            "en" : "You didn't choose a building / location",
            "fr" : "Vous n'avez pas choisi un bâtiment / endroit"
        },
        "make_request_no_urgency" : {
            "en" : "You didn't choose an urgency",
            "fr" : "Vous n'avez pas choisi l'urgence"
        },
        "make_request_no_description" : {
            "en" : "You didn't type a description!",
            "fr" : "Vous n'avez pas entré une description"
        },
        "make_request_no_title" : {
            "en" : "You didn't enter a title?",
            "fr" : "Vous n'avez entré un titre"
        },
        "completed_requests" : {
            "en" : "Completed Requests?",
            "fr" : "Requête Finie?"
        },
        
        


        "generic_error" : {
            "en" : "Oops something went wrong! - Please speak to the IT",
            "fr" : "Une erreur s'est produite, veuillez contacter le service informatique"
        }



        
    }
}