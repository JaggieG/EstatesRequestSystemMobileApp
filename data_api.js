export const getAllMyRequests = async (appInfo,callback) => {
    const baseURL = getBaseURL(appInfo)
    var myRequestEndpoint = 'getMyRequests'
    var completeURL = baseURL + myRequestEndpoint
    var email_address = getEmailAddress(appInfo)
    var JWT_Token = getJWTToken(appInfo)

    try {
        let response = await fetch(completeURL, {
            method: 'POST',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'application/json',
                'Authentication' : 'Bearer ' + JWT_Token
            },
            body: JSON.stringify({
                email_address: email_address
            })
        });
        
        let json = await response.json();
        callback(null, json)
    } catch(err) {
        console.log(err)
        callback(err,null)
        return err
    }


}

function getBaseURL(appInfo) {
    return appInfo.api_details.api_server_url + appInfo.api_details.api_path
}

function getEmailAddress(appInfo) {
    return appInfo.email_address
}

function getJWTToken(appInfo) {
    return appInfo.JWT_Token
}