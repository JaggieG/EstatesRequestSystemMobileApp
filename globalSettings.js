export const forceDevCreditionals = true
export const forced_dev_emailAdddress = 'jag@iglon.ch'
export const forced_dev_displayName = 'John Gerhardt'
export const forced_dev_JWTToken = 'FORCED_DEV'


export const getCurrentActiveLanguage = () => {
    return 'en'
}


