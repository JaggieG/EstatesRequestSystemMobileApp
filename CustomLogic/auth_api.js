import { Platform } from 'react-native';

import { 
  Linking,
  Alert
} from 'react-native';



import { getTranslatedMessage } from '../CustomLogic/messages.js'

import {
  forceDevCreditionals, 
  forced_dev_emailAdddress,
  forced_dev_displayName,
  forced_dev_JWTToken,
  force_dev_intSystemRole,
  force_settings_reload_at_restart,
} from './globalSettings.js'

import appInfoStore from './appInfoStore.js';
import { getAppInfo, clearStorage } from './storage';

import { defaultAppInfo } from './globalSettings';


export const authenticateMe = (appInfo, appInfoStore, refreshMe) => {
  // Go off an do Google oAuth
  Linking.getInitialURL().then(function(initialUrl) {
    
    var platform = Platform.OS

    const authURL = appInfo.api_details.api_server_url + appInfo.api_details.api_path + appInfo.api_details.authentication_endpoint
    if (forceDevCreditionals) {
    
      Alert.alert(getTranslatedMessage('forced_dev', appInfoStore))
        
      const api_details = defaultAppInfo.api_details
        
        appInfoStore.dispatch({
        type: "LOGIN",
        payload: { 
          email_address: forced_dev_emailAdddress, 
          display_name: forced_dev_displayName, 
          JWT_Token: forced_dev_JWTToken,
          api_details : api_details,
          int_SystemRole: force_dev_intSystemRole,
        }
      });
      
      // as we have some fake login data
      refreshMe()
    } else {
      if (initialUrl.indexOf("?")>-1){
        initialUrl = initialUrl.substring(0,initialUrl.indexOf("?"));
        }

      if (platform == 'web') {
        // We need to make an http call rather than a deep link
        const completeURL = authURL + '?returnURL=' + encodeURI(initialUrl)
        //Linking.openURL(completeURL)

        const api_details = defaultAppInfo.api_details
        Alert.alert(getTranslatedMessage('forced_dev',appInfoStore))
        appInfoStore.dispatch({
          type: "LOGIN",
          payload: { 
            email_address: forced_dev_emailAdddress, 
            display_name: forced_dev_displayName, 
            JWT_Token: forced_dev_JWTToken,
            api_details : api_details, 
            int_SystemRole: force_dev_intSystemRole,
          }
        });  
        refreshMe()
      } else {
        const completeURL = authURL + '?returnURL=' + encodeURI(initialUrl)
        console.log(completeURL)
        if(initialUrl) {
        
          Linking.openURL(completeURL)
        } else {
          Alert.alert(getTranslatedMessage('url_failure', appInfoStore))
        }
        
      } 
    }
      
  });  
}

export const processAuthReturn = (url_, appStoreInfo, refreshMe) => {
  const url = url_.url
  if (url) {
    if (url.indexOf("?retrieval_token") !== -1) {
        // now we have a  retreiva token we have 1 minute to use to to get our JWT token. We post it to the endpoint for security
        var regex = /[?&]([^=#]+)=([^&#]*)/g,
        params = {},
        match;
            while (match = regex.exec(url)) {
                params[match[1]] = match[2];
            }  

        const appInfo = appStoreInfo.getState()
        const retrieval_url = appInfo.api_details.api_server_url + appInfo.api_details.api_path + appInfo.api_details.retrieval_endpoint
          console.log(url)
        fetch(retrieval_url, {
          method: 'POST',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            retrieval_token: params["retrieval_token"],
          })
        }).then((response) => response.json())
          .then((json) => {
            console.log(json)
            var record_count = json.record_count

            if (record_count == 0) {
              Alert.alert(getTranslatedMessage('authentication_failure_noRecords', appInfoStore))
            } else {
              var the_record = json.data[0]
              appInfoStore.dispatch({
                type: "LOGIN",
                payload: { 
                  email_address: the_record.text_emailAddress, 
                  display_name: the_record.text_displayName,
                  JWT_Token: the_record.text_jwtToken,
                  int_SystemRole: the_record.int_SystemRole,
                }
              });
              
              
            }
            refreshMe()
          })
          .catch((error) => {
            console.error(error);
            refreshMe(!newState)
          });
        }
      }
}

export async function processAuthAtStartUp(appInfoStore, callback) {
 
  var appInfoFromStorage = await getAppInfo()
 
  // If there is nothing in the app Info Store then this is the first time the app launched - use the default
  if (force_settings_reload_at_restart) {  
    const local_state = appInfoStore.getState()
      const local_JWT_token = local_state.JWT_Token
      const local_api_details = JSON.stringify(local_state.api_details)
      const local_app_language = local_state.app_language 
      const local_display_name = local_state.display_name
      const local_email_address = local_state.email_address
      const local_intSystemRole = local_state.int_SystemRole

    const default_state = defaultAppInfo
      const default_JWT_token = default_state.JWT_Token
      const default_api_details = JSON.stringify(default_state.api_details)
      const default_app_language = default_state.app_language 
      const default_display_name = default_state.display_name
      const default_email_address = default_state.email_address
      const default_intSystemRole = default_state.int_SystemRole
      
      var changeFound = false
      if (local_api_details != default_api_details) {
        changeFound = true
      }

      if (local_app_language != default_app_language) {
        changeFound = true
      }

      if (local_intSystemRole != default_intSystemRole) {
        changeFound = true
      }
    
    if (changeFound) {
      appInfoStore.dispatch({
        type: "STARTUP",
        payload: defaultAppInfo
      });
      callback(true)
    } else {
      callback(true)
      // console.log('Global Settings Default State macthes with current state for:')
      // console.log(local_api_details)
      // console.log('=')
      // console.log(default_api_details)
      // console.log('LANG: ' + local_app_language + ' = ' + default_app_language) 
      // console.log('ROLE: ' + local_intSystemRole + ' = ' + default_intSystemRole) 
      
    }
    
  } else {
        if (appInfoFromStorage) {
            appInfoStore.dispatch({
              type: "STARTUP",
              payload: appInfoFromStorage
            });
          } else {
            appInfoStore.dispatch({
              type: "STARTUP",
              payload: defaultAppInfo
            });
          }
          callback(true)
    }
}
