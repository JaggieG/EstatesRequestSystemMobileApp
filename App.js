// imprt the required items from React native
import { 
  Dimensions, 
  Linking,
  View,
  Text,
  ActivityIndicator,
  StyleSheet,
} from 'react-native';

import { NavigationContainer} from '@react-navigation/native';
import React, { useEffect, useState } from 'react';

// Global device setupss

const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;

// Custom Components Info
import { CustomTabNavigatorComponent } from './CustomComponents/TabNavigator'
//import {ErrorView} from './CustomComponents/ErrorView'

//custom logic

import appInfoStore from './CustomLogic/appInfoStore'

import {processAuthReturn, processAuthAtStartUp} from './CustomLogic/auth_api' 



// Here is the main applicatoin
export default function App() {
  // This state allows the application to be refresh when a use is authentication
  const [refreshValue, _setRefreshValue] = useState(false); 

  // As we pull the current logged on user from the secure storage we need to wait for it to return
  // to be able to have access to the data.
  const [readyToLoad, setReadyToLoad] = useState(false)
  processAuthAtStartUp(appInfoStore, setReadyToLoad)

  const refreshMe = () => {
    _setRefreshValue(!refreshValue);
  };

  
  useEffect(() => {
    // Listening for the return URL when authenticating, we pass the url along with the store and a refreshMe function to refresh the root
    // nav controller
    
    Linking.addEventListener('url', function(url) {
      processAuthReturn(url, appInfoStore, refreshMe)
    })   
  }, []);

  if (readyToLoad) {
    return (

      <NavigationContainer refreshValue={refreshValue}>
         <CustomTabNavigatorComponent appInfoStore={appInfoStore} refreshMe={refreshMe}></CustomTabNavigatorComponent>
      </NavigationContainer>
    
  );
  } else {
    // this is so quick that probably willnever be seen, but put an activity indicator to be nice!
    return (

      <View style={baseStyles.baseView}>
          <ActivityIndicator size="large" />
      </View>
    
  );
  }
   
}

const baseStyles = StyleSheet.create({
  baseView : {
      flex: 1, 
      backgroundColor: '#FFF',
      justifyContent: 'center',
      alignItems: 'center' ,
  },
})