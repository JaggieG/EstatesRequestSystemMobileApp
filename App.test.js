import React from 'react'
import {render} from 'react-native-testing-library'
import App from './App'

import CustomTabNavigatorComponent from './CustomComponents/TabNavigator'
import ConnectionTabComponent from './CustomComponents/ConnectionTab'
import  MyRequestsTabComponent from './CustomComponents/MyRequestsTab'
import  MakeARequestTabComponent from './CustomComponents/MakeARequestTab'
import  SettingsTabComponent  from './CustomComponents/SettingsTab'
import  AssignedRequestTabComponent  from './CustomComponents/AssignedRequestsTab'

import "isomorphic-fetch";

// Extend Jest "expect" functionality with Testing Library assertions.
import '@testing-library/jest-dom'

import {getAllMyRequests,getSystemRole, getItemList,getUrgencyList, getRequestRecordCounts,getAllMyAssignedRequests} from './CustomLogic/data_api'
import {processAuthAtStartUp} from './CustomLogic/auth_api'
import { getTranslatedMessage } from './CustomLogic/messages'

import appInfoStore from './CustomLogic/appInfoStore'
import {forced_dev_emailAdddress, forced_dev_displayName, forced_dev_JWTToken} from './CustomLogic/globalSettings'

describe("Check data api", () => {
  const appInfo = appInfoStore.getState()

  // force the required dev info so the tests work
  appInfo.email_address = forced_dev_emailAdddress
  appInfo.display_name = forced_dev_displayName
  appInfo.JWT_Token = forced_dev_JWTToken


  it('getAllMyRequests return expected data', done => {
     getAllMyRequests(appInfo, true, function(err, dataReturn) {
       if (err) {
          done(true)
       } else {
  
        if (dataReturn.record_count >= 0 ) {
          done(false)
        } else {
          
          done(true)
        }
       } 
    })
  })


 it('getAllMyAssignedRequests returns expected data', done => {
  getAllMyAssignedRequests(appInfo, true, function(err, dataReturn) {
    if (err) {
       done(true)
    } else {

     if (dataReturn.record_count >= 0 ) {
      done(false)
     } else {
       
       done(true)
     }
    } 
 })
})



it('getRequestRecordCounts returns expected data', done => {
  getRequestRecordCounts(appInfo, true, function(err, dataReturn) {
    if (err) {  
       done(true)
    } else {

     if (dataReturn.assigned_requests >= 0  && dataReturn.my_requests>= 0) {
      done(false)
     } else {     
       done(true)
     }
    } 
 })
})



it('getUrgencyList returns expected data', done => {
  getUrgencyList(appInfo, function(err, dataReturn) {
    if (err) {
       done(true)
    } else {

     if (dataReturn.message.length > 0 ) {
       done(false)
     } else {
       done(true)
     }
    } 
 })
})



it('getItemList returns expected data', done => {
  getItemList(appInfo, function(err, dataReturn) {
    if (err) {
      done(true)
   } else {

    if (dataReturn.message.data.length > 0 ) {
      done(false)
    } else {
      done(true)
    }
   } 
 })
})

it('getSystemRole returns expected data', done => {
  getSystemRole(appInfo, function(err, dataReturn) {
    if (err) {
      done(true)
   } else {

    if (dataReturn.message >= 0 ) {
      done(false)
    } else {
      done(true)
    }
   } 
 })
})

})

describe("Check authentication api", () => {
  const appInfo = appInfoStore.getState()

  // force the required dev info so the tests work
  appInfo.email_address = forced_dev_emailAdddress
  appInfo.display_name = forced_dev_displayName
  appInfo.JWT_Token = forced_dev_JWTToken

    it('processAuthAtStartUp returns expected data', done => {
      processAuthAtStartUp(appInfoStore, function(data) {
        if (dataReturn) {
          done(false)
        } else {
          done(false)
        }
       
     })
    })
})


describe("Check translation class api", () => {
  const appInfo = appInfoStore.getState()


    it('getTranslatedMessage works for English', () => {
      
      // force language to English

      appInfoStore.dispatch({
        type: "LANG_UPDATE",
        payload: { 
            app_language : 'en',
        }
        });  


      var english_value = getTranslatedMessage('make_request_title', appInfoStore)
      expect(english_value).toBe('Title')
    })


    it('getTranslatedMessage works for French', () => {

      // force language to French

      appInfoStore.dispatch({
        type: "LANG_UPDATE",
        payload: { 
            app_language : 'fr',
        }
        });  

        var french_value = getTranslatedMessage('make_request_title', appInfoStore)
        expect(french_value).toBe('Titre')
    })
})

 describe("Functional Components", () => {
  it('<App /> should match snapshot', () => {
    const snap = render(<App />).toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('<Tab Navigtor /> should match snapshot', () => {
    const snap = render(<CustomTabNavigatorComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('<ConnectionTabComponent /> should match snapshot', () => {
    const snap = render(<ConnectionTabComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('<MyRequestsTabComponent /> should match snapshot', () => {
    const snap = render(<MyRequestsTabComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('<MakeARequestTabComponent /> should match snapshot', () => {
    const snap = render(<MakeARequestTabComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('<SettingsTabComponent /> should match snapshot', () => {
    const snap = render(<SettingsTabComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })
  it('<AssignedRequestTabComponent /> should match snapshot', () => {
    const snap = render(<AssignedRequestTabComponent />).toJSON()
    expect(snap).toMatchSnapshot()
  })
    
 });