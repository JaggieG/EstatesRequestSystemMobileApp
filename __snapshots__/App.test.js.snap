// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snap shot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Functional Components <App /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
